src/ordered_set_lang.ml

```diff
@@ -212,6 +223,52 @@ let standard =
 
 let field ?(default=standard) name = Sexp.Of_sexp.field name t ~default
 
+module Partial = struct
+  type ast =
+    ((Loc.t * Value.t list, String_with_vars.t) String_with_vars.Partial.t
+    , Ast.partial
+    ) Ast.t
+
+  type t = ast generic
+
+  let expand t ~dir ~files_contents ~(f : String_with_vars.t -> Value.t list) =
```

Comment:
What kind of signature for `eval_partial` are you consider considering? This?

```
val eval_partial
  :  t
  -> parse:(loc:Loc.t -> string -> value)
  -> dir:Path.t
  -> files_contents:Sexp.Ast.t Path.Map.t
  -> f:(String_with_vars.t -> Value.t list)
  -> standard:value list
  -> value list
```

Where `t` would be basically what I have as `Oredered_set_lang.Partial.t` now

I'd like to make things more efficient but this signature is quite busy already, and the stages don't really have well defined scope anymore.

We do have cases where we are expanding the OSL without any variables (for example in `parse_modules`). It doesn't seem so nice to complicate those uses. Also, I'm not entirely sure how we'd reconcile `Value.t`'s with `value`. We could do it via `parse (Value.to_string value)`, but this seems like what we're trying to avoid.

Reply:
I was thinking to have this in addition to the current evaluation functions. I forgot `value` was abstract, I suppose this function needs to go outside of the functor then. As a result, we shouldn't need the `parse` function, and maybe not the `dir` argument. This should simplify the signature.

Reply:
Okay, that's fine. We'll still have this annoying duplication between the eval's, but at least it will localized to the Osl module. I suppose I'll only implement the ordered variant of evaluation here, as the other isn't used with templates.

Reply:
seems good

Reply:
Okay, this is what I've done. As discussed, the comparison is done as a string which necessitates the `~dir` arg after all. It might also be just simpler to return `string list` from `expand`.